mitosheet/src/components/taskpanes/SnowflakeImport/SnowflakeImportTaskpane.tsx

```diff
@@ -239,7 +274,12 @@ const SnowflakeImportTaskpane = (props: SnowflakeImportTaskpaneProps): JSX.Eleme
                                 value={params.table_loc_and_warehouse.database || 'None available'}
                                 disabled={loadingAvailableOptionsAndDefaults}
                                 onChange={(newDatabase) => {
-                                    const newParams = getNewParams(params, newDatabase)
+                                    const newParams = getNewParams(
+                                        params, 
+                                        params.table_loc_and_warehouse.role, 
+                                        params.table_loc_and_warehouse.role, 
+                                        newDatabase
```

Comment:
Yeah. So this looks wrong -- but I also think it's probably wrong as a result of the types on this function -- all strings!

My thinking is that we should create a new function here that uses types similar to the stuff we use in graph, where there are partial updates, and we pass things by name (the reason we have this separate function is to set things to undefined, if I remember correctly). 

Thoughts?